### PR TITLE
Crosscompilation for s390x and ppc64le

### PR DESCRIPTION
**Pull request by fredericdalleau**
**- What I did**
Update build environment to provide cross compilation for linux/s390x and linux/ppc64le

**- How I did it**

Modify runtime-dev-cross-false and  runtime-dev-cross-true docker environments.
Use bullseye to provide libseccomp-dev as s390x and ppc64le in buster-backports are conflicting with other archs.

**- How to verify it**

```
$ make DOCKER_CROSSPLATFORMS="linux/s390x linux/ppc64le" cross
$ ls -l bundles/cross/linux/{s390x,ppc64le}-daemon
$ file bundles/cross/linux/{s390x,ppc64le}-daemon/dockerd-dev
```

SO FAR THE RESULTING BINARIES ARE UNTESTED

**- Description for the changelog**

Add cross compilation for s390x and ppc64le

**- A picture of a cute animal (not mandatory but encouraged)**

choupinou2

### CODE REVIEW

**Comment by thaJeztah** [comment]
Interesting failure (haven't seen that one before I think)

```
=== Failed
=== FAIL: pkg/archive TestTarUntarWithXattr (0.02s)
    archive_unix_test.go:288: assertion failed: string "/tmp/docker-test-untar-origin204062861/2 cap_block_suspend=ep\n" does not contain "= cap_block_suspend+ep": untar should have kept the 'security.capability' xattr
    archive_unix_test.go:288: assertion failed: string "/tmp/docker-test-untar-origin204062861/2 cap_block_suspend=ep\n" does not contain "= cap_block_suspend+ep": untar should have kept the 'security.capability' xattr
```

**Comment by thaJeztah** [comment]
Looks like there's some circular import, causing some things to fail:

```
=== Errors
import cycle not allowed in test
package github.com/docker/docker/quota (test)
	imports github.com/docker/docker/quota/helpers
	imports github.com/docker/docker/quota
```

**Comment by thaJeztah** [comment]
Ah; that's coming from https://github.com/moby/moby/pull/42172 (which is included here?) looks like that PR  isn't working

**Comment by fredericdalleau** [comment]
Ho yes, the cross build was failing on these architectures and it was exactly the error in https://github.com/moby/moby/pull/42172.
Is there a better way to use this PR?

**Comment by fredericdalleau** [comment]
Ha maybe, it's no longer necessary as I enable CGO after that. Let me check

**Comment by thaJeztah** [comment]
I think you need to add the new architectures to https://github.com/moby/moby/blob/a77317882d010b884a9101c6ad0b2d7db141082f/hack/make/cross#L15, so that CI builds them.

**Comment by fredericdalleau** [comment]
Ok I was unsure whether to set them default or not.

**Comment by thaJeztah** [comment]
Yeah, we could either pass it through from the Jenkinsfile, or set it by default. Probably enabling it by default is ok.

**Comment by thaJeztah** [comment]
```
objects: 100% (23/23), done.
[2021-06-17T20:02:33.002Z] Building: bundles/cross/windows/amd64/containerutility.exe
[2021-06-17T20:02:33.002Z] Cross building: bundles/cross/linux/ppc64le
[2021-06-17T20:02:33.002Z] Building: bundles/cross/linux/ppc64le-daemon/dockerd-dev
[2021-06-17T20:02:33.002Z] GOOS="linux" GOARCH="ppc64le" GOARM=""
[2021-06-17T20:02:33.002Z] # runtime/cgo
[2021-06-17T20:02:33.002Z] cgo: exec powerpc64le-linux-gnu-gcc: exec: "powerpc64le-linux-gnu-gcc": executable file not found in $PATH
script returned exit code 2

**Comment by fredericdalleau** [comment]
I used `make cross` on the host for testing this script.
The CI runs `hack/make.sh cross` from the `docker-dev` container.
The `docker-dev` is built from `runtime-dev-cross-false` as thus does not include the crossbuild packages.

**Comment by thaJeztah** [comment]
😞 looks like it's still missing something for some reason 😞 

```

[2021-06-22T10:29:26.616Z] Building: bundles/cross/windows/amd64/containerutility.exe
[2021-06-22T10:29:31.930Z] Cross building: bundles/cross/linux/ppc64le
[2021-06-22T10:29:31.930Z] Building: bundles/cross/linux/ppc64le-daemon/dockerd-dev
[2021-06-22T10:29:31.930Z] GOOS="linux" GOARCH="ppc64le" GOARM=""
[2021-06-22T10:29:32.507Z] # runtime/cgo
[2021-06-22T10:29:32.507Z] cgo: exec powerpc64le-linux-gnu-gcc: exec: "powerpc64le-linux-gnu-gcc": executable file not found in $PATH
```

**Comment by fredericdalleau** [comment]
The updated Jenkinsfile is not used. I issue a separate PR for it at https://github.com/moby/moby/pull/42551

**Comment by thaJeztah** [comment]
Oh 🤦 yes, Jenkinsfile updates are only used if the PR was opened by someone with write access on the repository.

**Comment by thaJeztah** [comment]
Opened https://github.com/moby/moby/pull/42552 with the first commit

**Comment by thaJeztah** [comment]
https://github.com/moby/moby/pull/42552 was merged; let me try if restarting Jenkins pick it up, otherwise you may have to do another rebase

**Comment by thaJeztah** [comment]
Cross compile is green now 🎉 

Screenshot 2021-06-23 at 14 58 48
Screenshot 2021-06-23 at 14 59 05

**Review by thaJeztah — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -127,16 +131,14 @@ ARG DEBIAN_FRONTEND
...
         apt-get update && apt-get install -y --no-install-recommends \
             libapparmor-dev:arm64 \
             libapparmor-dev:armel \
-            libapparmor-dev:armhf \
-            libseccomp-dev:arm64/buster-backports \
-            libseccomp-dev:armel/buster-backports \
-            libseccomp-dev:armhf/buster-backports
+            libapparmor-dev:ppc64el\
+            libapparmor-dev:s390x \
+            libapparmor-dev:armhf
```
Comment (tianon) [inline Dockerfile:141]: For consistency, I think these should be in the same (sorted) order as those above:

```suggestion
            libapparmor-dev:armhf \
            libapparmor-dev:s390x \
            libapparmor-dev:ppc64el
```

Comment (fredericdalleau) [reply Dockerfile]: Thanks, fixed now

**Review by tianon — APPROVED** [review]
:+1:

**Review by samuelkarp — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
we can ignore the libnetwork failure (unrelated)

**Comment by thaJeztah** [comment]
everything else completed and is green, so I'll merge this one

**Pull request merged**